### PR TITLE
add endpoints for assessment get and create

### PR DESCRIPTION
**Pull request by eckermania**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/eas/api/repository/v1/AssessmentRepository.java [hunk 1]
```diff
@@ -2,11 +2,23 @@
...
 
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface AssessmentRepository extends JpaRepository<AssessmentEntity, UUID> {
+    Optional<List<AssessmentEntity>> findByAssessmentTypeCode(String typeCode);
 
+    @Query("SELECT a FROM AssessmentEntity a WHERE a.sessionEntity.schoolYear = :schoolYear")
```
Comment (sumathi-thirumani) [inline api/src/main/java/ca/bc/gov/educ/eas/api/repository/v1/AssessmentRepository.java]: We don't need to include native queries; just referencing with _ will support. Reference: AssessmentStudentRepository.java

File: api/src/main/java/ca/bc/gov/educ/eas/api/validator/AssessmentValidator.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public class AssessmentValidator {
         this.sessionRepository = sessionRepository;
     }
 
-    public List<FieldError> validatePayload(Assessment assessment) {
+    public List<FieldError> validatePayload(Assessment assessment, boolean isCreateOperation) {
```
Comment (sumathi-thirumani) [inline api/src/main/java/ca/bc/gov/educ/eas/api/validator/AssessmentValidator.java:48]: We could leverage the groups concept; so can avoid extra flag. Reference:
@NotBlank(groups = OnUpdate.class, message = "assessmentStudentID cannot be null ")

**Review by sumathi-thirumani — DISMISSED** [review]

File: api/src/main/java/ca/bc/gov/educ/eas/api/endpoint/v1/AssessmentEndpoint.java [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+    @PreAuthorize("hasAuthority('SCOPE_READ_EAS_SESSIONS')")
+    @GetMapping("/{assessmentID}")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"), @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    @Tag(name= "Assessment Entity", description = "Endpoints for assessment entity.")
+    Assessment getAssessment(@PathVariable UUID assessmentID);
+
+    @GetMapping( "")
+    @PreAuthorize("hasAuthority('SCOPE_READ_EAS_SESSIONS')")
+    @Transactional(readOnly = true)
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"), @ApiResponse(responseCode = "404", description = "NOT FOUND")})
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/eas/api/endpoint/v1/AssessmentEndpoint.java]: Can't be a 404

File: api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/AssessmentService.java [hunk 4]
```diff
@@ -21,7 +22,31 @@
...
+            optionalAssessments = Optional.of(assessmentRepository.findAll());
+        }
+        return optionalAssessments.orElseThrow(() -> new EntityNotFoundException(AssessmentEntity.class, "typeCode", typeCode, "schoolYearEnd", schoolYearEnd));
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public AssessmentEntity createAssessment(AssessmentEntity assessmentEntity){
+        TransformUtil.uppercaseFields(assessmentEntity);
+        return assessmentRepository.save(assessmentEntity);
+    }
```
Comment (eckermania) [inline api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/AssessmentService.java:34]: one liner

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**